packages/client/wallets/smart-wallet/src/blockchain/wallets/service.ts

```diff
@@ -128,6 +129,41 @@ export class SmartWalletService {
                 address: entryPointVersion === "v0.6" ? ENTRYPOINT_ADDRESS_V06 : ENTRYPOINT_ADDRESS_V07,
             },
             kernelVersion,
+            existingSignerConfig: this.getSigner(signers),
         };
     }
+
+    private getSigner(signers: any[]): SignerData | undefined {
+        if (signers.length === 0) {
+            return undefined;
+        }
+
+        if (signers.length > 1) {
+            throw new CrossmintServiceError("Invalid wallet signer configuration. Please contact support");
+        }
+
+        return signers[0].signerData;
+    }
+}
+
+class AccountFactory {
+    constructor(private readonly eoa: EOAAccountService, private readonly passkey: PasskeyAccountService) {}
+
+    public get(
+        params: WalletCreationParams,
+        existingSignerConfig?: SignerData
+    ): Promise<{
+        signerData: SignerData;
+        account: KernelSmartAccount<EntryPoint, HttpTransport>;
+    }> {
+        if (isPasskeyParams(params) && (existingSignerConfig == null || existingSignerConfig?.type === "passkeys")) {
+            return this.passkey.get(params, existingSignerConfig);
+        }
+
+        if (existingSignerConfig == null || existingSignerConfig?.type === "eoa") {
+            return this.eoa.get(params as EOAWalletParams, existingSignerConfig);
+        }
+
+        throw new SmartWalletSDKError("Admin Mismatch");
```

Comment:
@bryan-eastwood3 I can implement this from the spec if you want!

Reply:
Yes pls!